oauth/src/lib.rs

```diff
@@ -11,23 +11,42 @@
 //! a spawned http server (mimicking Spotify's client), or manually via stdin. The latter
 //! is appropriate for headless systems.
 
-use log::{error, info, trace};
-use oauth2::basic::BasicTokenType;
-use oauth2::{
-    AuthUrl, AuthorizationCode, ClientId, CsrfToken, EndpointNotSet, EndpointSet,
-    PkceCodeChallenge, RedirectUrl, Scope, TokenResponse, TokenUrl, basic::BasicClient,
-};
-use oauth2::{EmptyExtraTokenFields, PkceCodeVerifier, RefreshToken, StandardTokenResponse};
-use std::io;
-use std::sync::mpsc;
-use std::time::{Duration, Instant};
 use std::{
-    io::{BufRead, BufReader, Write},
+    io::{self, BufRead, BufReader, Write},
     net::{SocketAddr, TcpListener},
+    sync::mpsc,
+    time::{Duration, Instant},
 };
+
+use oauth2::{
+    AuthUrl, AuthorizationCode, ClientId, CsrfToken, EmptyExtraTokenFields, EndpointNotSet,
+    EndpointSet, PkceCodeChallenge, PkceCodeVerifier, RedirectUrl, RefreshToken, Scope,
+    StandardTokenResponse, TokenResponse, TokenUrl, basic::BasicClient, basic::BasicTokenType,
+};
+
+use log::{error, info, trace};
 use thiserror::Error;
 use url::Url;
 
+// TLS Feature Validation
+//
+// These compile-time checks are placed in the oauth crate rather than core for a specific reason:
+// oauth is at the bottom of the dependency tree (even librespot-core depends on librespot-oauth),
+// which means it gets compiled first. This ensures TLS feature conflicts are detected early in
+// the build process, providing immediate feedback to users rather than failing later during
+// core compilation.
+//
+// The dependency chain is: workspace -> core -> oauth
+// So oauth's feature validation runs before core's, catching configuration errors quickly.
```

Comment:
I looked into why core dependence on `oauth` and the reason is quite silly but make somewhat sense. For most errors we implement `From<CustomError> for Error`, the same happens for `oauth`.

So we could drop the implementation of `From<OAuthError>` in core then this dependency would fall away. Might be a bit less convenient but would resolve the dependency issue here.

Reply:
We could also look into separating more common type/struct/enum out of core that might be used in other packages, but don't need core (`Session`) in particular. Not for this PR, but just a general thought that came up.

Reply:
Good one. It may be worth doing that, removing the `From` implementation if it saves pulling in that dependency.

Reply:
Yeah, that we don't intend to change. But maybe extracting something of core so that core doesn't depend on oauth would probably be better in the long run, as core itself doesn't use anything out of oauth. But a topic outside of this PR in the end^^

Reply:
I quickly tried removing the `From<OAuthError> for Error` implementation. Strangely:
1. the build is successful without it (dead code?)
2. the `compile_error` still needs to live in `oauth/src/lib.rs` to work

Maybe I'm missing something obvious, but as this seems to require more investigation I'll also leave it to a later PR.